tonami/pitch_process.py

```diff
@@ -96,10 +96,52 @@ def get_voice_activity(pitch_contour):
 
     return voiced
 
+def get_nan_idx(arr):
+    """
+    Helper to handle indices and logical indices of NaNs.
+
+    Args:
+        arr: 1d numpy array with possible NaNs
+    Returns:
+        nans: logical indices of NaNs (an array of the same size as arr, 
+            which marks the location of NaNs with 1, and all others with 0)
+        index: a function that takes an array with 1s representing NaNs, 
+            and returns the indexes of nans
+    Example:
+        >>> # linear interpolation of NaNs
+        >>> arr = np.array([3,np.nan,4,np.nan,5)
+
+        >>> nans, index= nan_helper(arr)
+        >>> # nans would be [0, 1, 0, 1, 0]
+        >>> # index(nans) would return [1, 3]
+        >>> arr[nans]= np.interp(index(nans), index(~nans), arr[~nans])
```

Comment:
might as well convert this fully to a doctest format (it's basically 90% of the way there)

https://docs.python.org/3/library/doctest.html

Reply:
lol